src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -23,52 +24,172 @@
 
 (def EnvironmentClassesFileEntry
   "Schema for an individual file entry which is part of the return payload
-  for an environment_classes request"
+  for an environment_classes request."
   {:path schema/Str
-   :classes [Map]})
+   schema/Keyword schema/Any})
 
 (def EnvironmentClassesInfo
-  "Schema for the return payload an environment_classes request"
+  "Schema for the return payload an environment_classes request."
   {:name schema/Str
    :files [EnvironmentClassesFileEntry]})
 
+(defn sort-nested-java-maps
```

Comment:
The approach of sorting keys within sub-maps of the response prior to calculating a UTF-8 byte hash of the JSON-serialized object follows the approach that PuppetDB uses - see https://github.com/puppetlabs/puppetdb/blob/7f7191c883ab138dcdae3e2c06581c6c0cf9f1cc/src/puppetlabs/puppetdb/scf/hash.clj#L14-L21.